src/libs/Violations/ViolationsUtils.ts

```diff
@@ -0,0 +1,69 @@
+import reject from 'lodash/reject';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+import {PolicyCategories, PolicyTags, Transaction, TransactionViolation} from '@src/types/onyx';
+import possibleViolationsByField, {ViolationField} from './possibleViolationsByField';
+
+const ViolationsUtils = {
+    getViolationForField(transactionViolations: TransactionViolation[], field: ViolationField, translate: (key: string) => string): string[] {
+        return transactionViolations.filter((violation) => possibleViolationsByField[field]?.includes(violation.name)).map((violation) => translate(violation.name));
+    },
+
+    getViolationsOnyxData(
+        /** The transaction to check for policy violations. */
+        transaction: Transaction,
+        /** An array of existing transaction violations. */
+        transactionViolations: TransactionViolation[],
+        /** Indicates if the policy requires tags. */
+        policyRequiresTags: boolean,
+        /** Collection of policy tags and their enabled states. */
+        policyTags: PolicyTags,
+        /** Indicates if the policy requires categories. */
+        policyRequiresCategories: boolean,
+        /** Collection of policy categories and their enabled states. */
+        policyCategories: PolicyCategories,
+    ): {
```

Comment:
This is not how we document method parameters, right? We can use js docs style for these params, but with their names and types they're pretty obvious, so I suggest just having a short description of what this function does, eg
```suggestion
    /**
     *  Computes an updated array of transaction violations for the given transaction
     */
    getViolationsOnyxData(
        transaction: Transaction,
        transactionViolations: TransactionViolation[],
        policyRequiresTags: boolean,
        policyTags: PolicyTags,
        policyRequiresCategories: boolean,
        policyCategories: PolicyCategories,
    ): {
```